File: Heydesk.Server/Domains/Agent/AgentValidator.cs
using FluentValidation;
using Heydesk.Server.Data.Models;

namespace Heydesk.Server.Domains.Agent;

public class CreateAgentValidator : AbstractValidator<CreateAgentRequest>
{
    public CreateAgentValidator()
    {
        RuleFor(a => a.Name)
            .NotEmpty()
            .MinimumLength(2)
            .MaximumLength(100);
        RuleFor(a => a.Description)
            .MaximumLength(500);
        RuleFor(a => a.SystemPrompt)
            .NotEmpty()
            .MinimumLength(10)
            .WithMessage("System prompt must be at least 10 characters");
        RuleFor(a => a.Type)
            .IsInEnum();
    }
}

File: Heydesk.Client/src/lib/validators/agent.validator.ts
import { z } from 'zod';
import { AgentType } from '@/lib/types/agent';

export const createAgentSchema = z.object({
  name: z
    .string()
    .min(2, "Name must be at least 2 characters")
    .max(100, "Name must be less than 100 characters"),
  description: z
    .string()
    .max(500, "Description must be less than 500 characters"),
  systemPrompt: z
    .string()
    .min(10, "System prompt must be at least 10 characters"),
  type: z.enum(AgentType),
});


